rust/libnewsboat/src/fmtstrformatter/mod.rs

```diff
@@ -353,6 +353,40 @@ mod tests {
         assert_eq!(fmt.do_format("%?c?asdf?", 0), "asdf");
     }
 
+    #[test]
+    fn t_do_format_keeps_wide_characters_within_specified_width() {
```

Comment:
These four tests have to be ported to C++, to ensure that FFI bindings don't change the behaviour and don't introduce any bugs.